src/flint/fmpq.jl

```diff
@@ -115,6 +115,8 @@ is_zero(a::QQFieldElemOrPtr) = is_zero(_num_ptr(a))
 
 is_unit(a::QQFieldElem) = !iszero(a)
 
+is_nilpotent(a::QQFieldElem) = iszero(a)
```

Comment:
I am not sure we really need or want these methods here -- I would expect this to be handled by a generic method in AA, something like this:
```
function is_nilpotent(a::T) where {T <: RingElem}
  is_domain_type(parent_type(T)) && return izero(a)
  throw(NotImplementedError(:is_nilpotent, a))
end
```

In general I think it would be better to first have the generic code in AA ready. BTW you mentioned again during the call today the problem of dealing with infinite series -- so I'll also mention again that this is not necessary for now -- the above method takes care of it well enough for now.

So a first PR just adding e.g. that method and perhaps a few tests would already be a good start. Adding `is_unit` methods can then come in one more further PRs. And then once AA is ready and released we can add missing or optimized bits in Nemo.